pallets/xcmp-queue/src/lib.rs

```diff
@@ -538,26 +538,24 @@ impl<T: Config> Pallet<T> {
 			XcmpMessageFormat::ConcatenatedEncodedBlob => {
 				while !remaining_fragments.is_empty() {
 					last_remaining_fragments = remaining_fragments;
-					match <Vec<u8>>::decode_all(&mut remaining_fragments) {
-						Ok(blob) if remaining_fragments.len() < last_remaining_fragments.len() => {
-							let weight = max_weight - weight_used;
-							match Self::handle_blob_message(sender, sent_at, blob, weight) {
-								Ok(used) => weight_used = weight_used.saturating_add(used),
-								Err(true) => {
-									// That message didn't get processed this time because of being
-									// too heavy. We leave it around for next time and bail.
-									remaining_fragments = last_remaining_fragments;
-									break
-								},
-								Err(false) => {
-									// Message invalid; don't attempt to retry
-								},
-							}
-						},
-						_ => {
-							debug_assert!(false, "Invalid incoming blob message data");
-							remaining_fragments = &b""[..];
-						},
+
+					if let Ok(blob) = <Vec<u8>>::decode(&mut remaining_fragments) {
```

Comment:
Do we not check for `remaining_fragments.len() < last_remaining_fragments.len()` anymore here?

Reply:
No. We will for sure consume something and that will then either return `Ok()` or `Err()`. If there is an error we abort.

`remaining_fragments.len() < last_remaining_fragments.len()` this check was also completely wrong in the old code and was always returning `false.

Reply:
Hold on, the code was always returning false previously because we used decode_all, which doesn't advance the cursor in remaining_fragments. If we do check it now, then we should expect it to return true most of the time.

The only time where I can see that it would not return true is if remaining_fragments is empty, and we keep on trying to decode it. I think we could instead make a debug_assert here to ensure that we always decode something.

Reply:
> not return true is if remaining_fragments is empty

`decode` would return an error when `remaining_fragments` is empty, because that is no valid encoded vec.